shared/features/invoice/handlers/fetch-font-info.ts

```diff
@@ -0,0 +1,125 @@
+import { franc } from "franc";
+
+interface Font {
+  src: string;
+  fontWeight?: number;
+  format?: string;
+}
+interface FontInfo {
+  family: string;
+  fonts: Font[];
+}
+
+const defaultFont = {
+  family: "GT Walsheim",
+  fonts: [
+    { src: `${process.env.NEXT_PUBLIC_METADATA_ASSETS_S3_BUCKET}/GT-Walsheim-Pro-Thin.ttf`, fontWeight: 100 },
+    { src: `${process.env.NEXT_PUBLIC_METADATA_ASSETS_S3_BUCKET}/GT-Walsheim-Pro-Ultra-Light.ttf`, fontWeight: 200 },
+    { src: `${process.env.NEXT_PUBLIC_METADATA_ASSETS_S3_BUCKET}/GT-Walsheim-Pro-Light.ttf`, fontWeight: 300 },
+    { src: `${process.env.NEXT_PUBLIC_METADATA_ASSETS_S3_BUCKET}/GT-Walsheim-Pro-Regular.ttf`, fontWeight: 400 },
+    { src: `${process.env.NEXT_PUBLIC_METADATA_ASSETS_S3_BUCKET}/GT-Walsheim-Pro-Medium.ttf`, fontWeight: 500 },
+    { src: `${process.env.NEXT_PUBLIC_METADATA_ASSETS_S3_BUCKET}/GT-Walsheim-Pro-Bold.ttf`, fontWeight: 700 },
+    { src: `${process.env.NEXT_PUBLIC_METADATA_ASSETS_S3_BUCKET}/GT-Walsheim-Pro-Ultra-Bold.ttf`, fontWeight: 800 },
+    { src: `${process.env.NEXT_PUBLIC_METADATA_ASSETS_S3_BUCKET}/GT-Walsheim-Pro-Black.ttf`, fontWeight: 900 },
+  ],
+};
+
+async function fetchFontInfo(url: string): Promise<FontInfo> {
+  try {
+    const response = await fetch(url);
+    const cssContent = await response.text();
+
+    const fontFamilyMatch = cssContent.match(/font-family: '(.+?)';/);
+    const srcMatch = cssContent.match(/src: url\((.+?)\)/);
+
+    if (fontFamilyMatch && srcMatch) {
+      return {
+        family: fontFamilyMatch[1],
+        fonts: [{ src: srcMatch[1], format: "truetype" }],
+      };
+    }
+
+    throw new Error("Font information not found");
+  } catch (error) {
+    console.error("Failed to fetch font info:", error);
```

Comment:
Do we need logs here ?

Reply:
yes to easily find the error context on server